src/Containers/SparseAxisArray.jl

```diff
@@ -122,14 +122,49 @@ function Base.getindex(
     return getindex(d, idx...)
 end
 
-function Base.getindex(d::SparseAxisArray{T,N}, idx...) where {T,N}
+function Base.getindex(d::SparseAxisArray{T,N,K}, idx...) where {T,N,K}
     if length(idx) < N
         throw(BoundsError(d, idx))
     end
-    _colon_error(idx...)
+    if _is_slice(first(keys(d.data)), idx)
```

Comment:
`first(keys(d.data))` will fail if the dictionary is empty

Reply:
I've updated the code to use a generated function, which has the nice side-effect of ensuring type stability of the result if the filter is empty.